path_roles.go

```diff
@@ -307,6 +338,88 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 	return resp, nil
 }
 
+func (b *azureSecretBackend) createPersistedApp(ctx context.Context, req *logical.Request, role *roleEntry, name string) error {
+	c, err := b.getClient(ctx, req.Storage)
+	if err != nil {
+		return err
+	}
+
+	if role.ManagedApplicationObjectID != "" {
+		// Unassign roles
+		if err := c.unassignRoles(ctx, role.RaIDs); err != nil {
```

Comment:
nit: might be worth factoring common functions for unassigning roles + removing group memberships and assigning roles + adding group memberships. Might resolve some duplicated code in this function as well as `pathRoleDelete`